src/renderer/helpers/utils.js

```diff
@@ -9,35 +9,35 @@ import router from '../router/index'
 // https://support.google.com/youtube/answer/11585688#change_handle
 export const CHANNEL_HANDLE_REGEX = /^@[\w.-]{3,30}$/
 
+const PUBLISHED_TEXT_REGEX = /(\d+)\s?([a-z]+)/i
+/**
+ * @param {string} publishedText
+ */
 export function calculatePublishedDate(publishedText) {
   const date = new Date()
   if (publishedText === 'Live') {
     return publishedText
   }
 
-  const textSplit = publishedText.split(' ')
-
-  if (textSplit[0].toLowerCase() === 'streamed') {
-    textSplit.shift()
-  }
+  const match = publishedText.match(PUBLISHED_TEXT_REGEX)
 
-  const timeFrame = textSplit[1]
-  const timeAmount = parseInt(textSplit[0])
+  const timeFrame = match[2]
+  const timeAmount = parseInt(match[1])
   let timeSpan = null
 
-  if (timeFrame.indexOf('second') > -1) {
+  if (timeFrame.startsWith('second') || timeFrame === 's') {
     timeSpan = timeAmount * 1000
-  } else if (timeFrame.indexOf('minute') > -1) {
+  } else if (timeFrame.startsWith('minute') || timeFrame === 'm') {
     timeSpan = timeAmount * 60000
-  } else if (timeFrame.indexOf('hour') > -1) {
+  } else if (timeFrame.startsWith('hour') || timeFrame === 'h') {
     timeSpan = timeAmount * 3600000
-  } else if (timeFrame.indexOf('day') > -1) {
+  } else if (timeFrame.startsWith('day') || timeFrame === 'd') {
     timeSpan = timeAmount * 86400000
-  } else if (timeFrame.indexOf('week') > -1) {
+  } else if (timeFrame.startsWith('week') || timeFrame === 'w') {
     timeSpan = timeAmount * 604800000
-  } else if (timeFrame.indexOf('month') > -1) {
+  } else if (timeFrame.startsWith('month') || timeFrame === 'mo') {
     timeSpan = timeAmount * 2592000000
-  } else if (timeFrame.indexOf('year') > -1) {
+  } else if (timeFrame.startsWith('year') || timeFrame === 'y') {
```

Comment:
Have you been able to test all of these? I saw a screenshot in the invidious matrix room that had some different values. (yr, wk, mo)
![Screenshot_20230527_163536_org schabi newpipe debug cronetasdownloadnetworkstack_20230527111314](https://github.com/FreeTubeApp/FreeTube/assets/78101139/914cda4a-0a2a-4b91-9560-2a408daa7851)